### ISSUE TITLE
Reader().map() fails to limit properly the number of parallel instances of a function.

### ISSUE THREAD

**Issue by slawo** [kriskowal/q-io#126]
When setting the limit to 1 the given function is executed twice in parallel:

Test case:

```
  var Q = require('q');
  var Reader = require("q-io/reader");
  var data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
  var count = 0;
  var func = function(d) {
    var deferred = Q.defer();
    console.log("+ d", d, "count:", ++count)
    setTimeout(function() {
      console.log("-d", d, "count:", --count);
      deferred.resolve(d);
    }, 200);
    return deferred.promise;
  }
  var runOnAll = new Reader(data).map(func, null, 1).all();
```

The resulting output:

```
+ d 1 count: 1
+ d 2 count: 2
-d 1 count: 1
-d 2 count: 0
+ d 3 count: 1
+ d 4 count: 2
-d 3 count: 1
-d 4 count: 0
+ d 5 count: 1
+ d 6 count: 2
-d 5 count: 1
-d 6 count: 0
+ d 7 count: 1
+ d 8 count: 2
-d 7 count: 1
-d 8 count: 0
+ d 9 count: 1
+ d 10 count: 2
-d 9 count: 1
-d 10 count: 0
```

q: 1.1.2
q-io: 2.0.6

**Comment by slawo** [comment]
Unless my understanding of the function `.map(func, null, 1)` is wrong the resulting behavior should be only one instance of `func` being executed at a time.

Tried with q: 2.0.2 to the same result.

**Comment by kriskowal** [comment]
Thank you!

**Comment by hthetiot** [comment]
Could be related to collections, we removed it
